### DISCUSSION TITLE
tincacms: not found when deploying in Netlify

### DISCUSSION THREAD

**Discussion by growdigital** [tinacms/tinacms#4424] [Help] [answered]
I've just reinstalled Tina after [previously messing up formatting](https://github.com/tinacms/tinacms/discussions/4421) on a Hugo install.

It works fine locally but when I deploy to Netlify I get the error `/bin/sh: 1: tincacms: not found`

I'm using the same commands as when the project initially worked ie `yarn project-setup && yarn build` and have tried various permuations (npm, install etc) but with no joy.

The full snippet is here: https://gist.github.com/growdigital/9339dc5e5b9edcf167f9b909df7b9bf3

It is guaranteed to be a Glaring User Error, any light shed on this gratefully received. Thank you :)

PS GitHub project https://github.com/growdigital/hopegarden.uk

**Comment by kldavis4** [comment]
Probably you are missing the node dependencies. I am not familiar with netlify but likely just changing the build command to `yarn && yarn project-setup && yarn build` will resolve.

**Reply by growdigital** [reply]
Thank you. Progress, in that's a way more verbose error message:
```
8:04:19 PM: warning " > tinacms@1.5.28" has unmet peer dependency "react@>=16.14.0".
```
etc.
Full gist: https://gist.github.com/growdigital/8432b1dbcdf5f82d3fb81ab54390d2c0

I don't know much about node/npm/yarn etc., maybe there's a difference of opinion between npm & yarn? I'll try using yarn for local dev, current command is `npm run dev`

**Reply by growdigital** [reply]
Even more interesting. Installed yarn, then get this warning:
```
hopegarden.uk [main] ⚡  yarn
…
➤ YN0000: ┌ Post-resolution validation
➤ YN0002: │ bookworm-hugo@workspace:. doesn't provide react (p34ce0), requested by tinacms.
➤ YN0002: │ bookworm-hugo@workspace:. doesn't provide react-dom (p95ee7), requested by tinacms.
➤ YN0086: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code.
```
Full gist: https://gist.github.com/growdigital/2d943db72267be185724a547b86eeef0

When I try `yarn dev`, I get a similar error message to the netlify one
```
Build failed with 242 errors:
.yarn/__virtual__/@floating-ui-react-dom-virtual-0b99391b84/3/.yarn/berry/cache/@floating-ui-react-dom-npm-2.0.5-930191328a-10c0.zip/node_modules/@floating-ui/react-dom/dist/floating-ui.react-dom.esm.js:3:23: ERROR: Could not resolve "react"
```
Full gist: https://gist.github.com/growdigital/5aaa0c3b3b68e0eb6b3378721c880444

Really feels like this is a "I don't know enough about node" problem! Thanks for looking :)

**Reply by growdigital** [reply]
FWIW
```
☁  hopegarden.uk [main] ⚡  yarn explain peer-requirements p95ee7
The yarn explain peer-requirements command doesn't support this warning type yet.
☁  hopegarden.uk [main] ⚡  yarn explain peer-requirements p34ce0
The yarn explain peer-requirements command doesn't support this warning type yet.
```

**Reply by growdigital** [reply]
I used `yarn add` to add `react` and `react-dom` to `package.json`. Now yarn works fine locally. Added `yarn.lock` to repository, the Netlify error messages are way reduced but still getting 
```
11:41:41 PM: /bin/sh: 1: tincacms: not found
11:41:41 PM: error Command failed with exit code 127. (https://ntl.fyi/exit-code-127)
```
Full gist: https://gist.github.com/growdigital/1b4b5c48967acfdf4b03084177150f45

**Comment by kldavis4** [comment, marked as answer]
Sorry, I missed the root cause - `tincacms` != `tinacms`

**Reply by growdigital** [reply]
OMG thank you and I am so sorry! I've been going through the logs trying to figure it out and I didn't check for typos :facepalm:
